File: frontend/src/types/batch.ts
export type BatchJobStatus = "queued" | "running" | "complete" | "failed";
export type JobResultStatus = "pending" | "success" | "failed" | "skipped";

export interface BatchJob {
  id: string;
  filename: string;
  status: BatchJobStatus;
  total_records: number;
  processed_records: number;
  successful_records: number;
  failed_records: number;
  unclear_records: number;
  progress_percentage: number;
  celery_task_id: string | null;
  uploaded_at: string;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

export interface JobResult {
  id: string;
  row_number: number;
  status: JobResultStatus;
  error_message: string | null;
  raw_csv_row: Record<string, string>;
  verification: {
    id: string;
    search_id: string;
    status: string;
    full_name: string;
    company_name: string;
    confidence_score: number;
    confidence_level: string;
    created_at: string;
  } | null;
}

File: backend/app/schemas/batch.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import BatchJobStatus, JobResultStatus
from app.schemas.verification import VerificationSummary


class BatchJobResponse(BaseModel):
    id: UUID
    filename: str
    status: BatchJobStatus
    total_records: int
    processed_records: int
    successful_records: int
    failed_records: int
    unclear_records: int
    progress_percentage: int
    celery_task_id: str | None = None
    uploaded_at: datetime
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class JobResultResponse(BaseModel):
    """Per-row result returned by GET /{job_id}/results."""
    id: UUID
    row_number: int
    status: JobResultStatus
    error_message: str | None
    raw_csv_row: dict
    verification: VerificationSummary | None
